src/main/java/org/icgc/argo/program_service/controller/ProgramController.java

```diff
@@ -137,4 +139,184 @@ public ResponseEntity<List<ProgramsResponseDTO>> listPrograms(
+    authorizationService.requireProgramAdmin(shortName, authorization);
+    val users = serviceFacade.listUsers(shortName);
+    if (users != null && !users.getUserDetailsList().isEmpty()) {
+      return new ResponseEntity<>(
+          grpc2JsonConverter.prepareListUsersResponse(users).getUserDetails(), HttpStatus.OK);
+    } else {
+      return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+    }
+  }
+
+  @DeleteMapping(value = "/users")
+  public ResponseEntity<RemoveUserResponseDTO> removeUser(
+      @Parameter(hidden = true) @RequestHeader(value = "Authorization", required = true)
+          final String authorization,
+      @RequestBody RemoveUserRequestDTO removeUserRequestDTO)
+      throws IOException {
+    authorizationService.requireProgramAdmin(
+        removeUserRequestDTO.getProgramShortName(), authorization);
+    RemoveUserRequest request =
+        grpc2JsonConverter.fromJson(
+            grpc2JsonConverter.getJsonFromObject(removeUserRequestDTO), RemoveUserRequest.class);
+    val users = serviceFacade.removeUser(request);
+    if (users != null) {
+      return new ResponseEntity<>(
+          grpc2JsonConverter.prepareRemoveUserResponse(users), HttpStatus.OK);
+    } else {
+      return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+    }
+  }
+
+  @PutMapping(value = "/users")
+  public void updateUser(
+      @Parameter(hidden = true) @RequestHeader(value = "Authorization", required = true)
+          final String authorization,
+      @RequestBody UpdateUserRequestDTO updateUserRequestDTO) {
+    authorizationService.requireProgramAdmin(updateUserRequestDTO.getShortName(), authorization);
+    UpdateUserRequest request;
+    try {
+      request =
+          grpc2JsonConverter.fromJson(
+              grpc2JsonConverter.getJsonFromObject(updateUserRequestDTO), UpdateUserRequest.class);
+      serviceFacade.updateUser(request);
+    } catch (NotFoundException | IOException e) {
+      log.error("Exception throw in joinProgram: {}", e.getMessage());
+      throw new NotFoundException("User not found");
+    }
+  }
+
+  @GetMapping(value = "/joinProgramInvite")
+  public ResponseEntity<GetJoinProgramInviteResponseDTO> getJoinProgramInvite(
+      @Parameter(hidden = true) @RequestHeader(value = "Authorization", required = true)
+          final String authorization,
+      @RequestBody GetJoinProgramInviteRequestDTO getJoinProgramInviteRequestDTO) {
+    val invitation =
+        serviceFacade.getInvitationById(
+            UUID.fromString(getJoinProgramInviteRequestDTO.getInviteId()));
+    GetJoinProgramInviteResponseDTO getJoinProgramInviteResponseDTO =
+        new GetJoinProgramInviteResponseDTO();
+    getJoinProgramInviteResponseDTO.setInvitation(
+        grpc2JsonConverter.prepareGetJoinProgramInviteResponse(invitation));
+    return new ResponseEntity<GetJoinProgramInviteResponseDTO>(
+        getJoinProgramInviteResponseDTO, HttpStatus.OK);
+  }
+
+  @GetMapping(value = "/listCancers")

```

Comment:
Can we switch to standard REST syntax? `GET /cancers`

Reply:
Updated as per review comment